### ISSUE TITLE
feat: add aria property to action

### ISSUE THREAD

**Issue by colinhale1** [Esri/calcite-design-system#10706]
**monday.com sync:** #18048550887

### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

Original Summary:
> During our recent accessibility audit, we discovered that the calcite-action button is read out as a "toggle button" when using a screen reader. @driskull mentioned in a meeting that "adding type to action makes sense, like 'toggle', 'button', 'expand-toggle'." We request the ability to control the "type" as suggested by Matt for calcite-action components.

### Actual Behavior

Cannot set Aria properties on Action.

Original
> Screen readers read calcite-action components as "toggle-button".

### Expected Behavior

Can set Aria properties on Action via an `aria` property.

Original
> Default behavior for a calcite-action should be read by a screen reader as "button"

### Reproduction Sample

https://next.gha.afd.arcgis.com/calcite-design-system/components/action/

### Reproduction Steps

Original Repro
> 1. Go to the calcite example for action bar:https://developers.arcgis.com/calcite-design-system/components/action-bar/
> 2. Put screen focus on button bar.
> 3. Start up screen reader(ctr+F5 or cmd+F).
> 4. Tab to a new calcite-action in the action bar.
> 5. Observe that button is read as "Toggle button"

### Reproduction Version

Current

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Business/Community Analyst

**Comment by driskull** [comment]
related to #10411 and #7003. Should be tackled together.

**Comment by driskull** [comment]
@geospatialem regarding the `toggle` state for action; do we need to have a "mixed" state? See "mixed" state here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-pressed#description

> Imagine a button that controls the behavior of a group of checkboxes. If some checkboxes are checked and others are not, the button could have aria-pressed="mixed" to reflect this indeterminate state.

**Comment by geospatialem** [comment]
> [@geospatialem](https://github.com/geospatialem) regarding the `toggle` state for action; do we need to have a "mixed" state? See "mixed" state here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-pressed#description
> 
> > Imagine a button that controls the behavior of a group of checkboxes. If some checkboxes are checked and others are not, the button could have aria-pressed="mixed" to reflect this indeterminate state.

@driskull As long as you agree, that seems in alignment with the pressed state, where there are different values represented, also mentioned on [DitialA11y](https://www.digitala11y.com/aria-pressed-state/#:~:text=A%20value%20of%20mixed%20means%20that%20the%20values%20of%20more%20than%20one%20item%20controlled%20by%20the%20button%20do%20not%20all%20share%20the%20same%20value.).

**Comment by driskull** [comment]
@geospatialem yeah I think that makes sense. We may need to consult design on how a "mixed" pressed state makes the component look different though. It's not like a checkbox where we can show the dashed checkbox.

@ashetland @SkyeSeitz any idea on how a "mixed" state pressed action could be visually different from one that is just pressed/active?

**Comment by ashetland** [comment]
Is `mixed` required? I'm not sure of Actions being used in this way:

> A value of `mixed` means that the values of more than one item controlled by the button do not all share the same value.

If this is indeed required, we'd need to research how to communicate that state. Action's current `:active` styling does not take a mixed state into account.

**Comment by driskull** [comment]
@geospatialem can you comment on above? 

It may be possible an `action` represents multiple items which may nor may not all be selected. correct? If we can think of an action in a toolbar usage there may potentially be use cases where an action represents selection of multiple items outside of the toolbar.

**Comment by driskull** [comment]
Maybe we can hold off on the `mixed` value for now until a good use case comes up.

**Comment by geospatialem** [comment]
> Maybe we can hold off on the `mixed` value for now until a good use case comes up.

This sounds like a good approach. 💪🏻

**Comment by driskull** [comment]
So the route we could take is as follows:

- add a `pressed: boolean;` property and in the future introduce a `mixed` value as `pressed: boolean | "mixed";`
- add a `selected: boolean;` property and in the future introduce a `indeterminate` property as `indeterminate: boolean;`

@jcfranco @geospatialem do you have a preference here?

**Comment by geospatialem** [comment]
> So the route we could take is as follows:
> 
> * add a `pressed: boolean;` property and in the future introduce a `mixed` value as `pressed: boolean | "mixed";`
>   
>   * consistent with `aria-pressed` property.
> * add a `selected: boolean;` property and in the future introduce a `indeterminate` property as `indeterminate: boolean;`
>   
>   * consistent with `checkbox` component although it has a `checked` property instead of `selected`
> 
> [@jcfranco](https://github.com/jcfranco) [@geospatialem](https://github.com/geospatialem) do you have a preference here?

I like the first option, it seems more in alignment with `aria-pressed`, and seems clear with `pressed`. Whereas the `selected` and `indeterminate` option seems like we'd need to provide additional context and doc on, especially with the `checked` vs `selected` context. But will defer to you and @jcfranco.

**Comment by driskull** [comment]
@geospatialem @jcfranco what do you think for the expand type? `type: "expand";` or `type: "expand-toggle`? 

```js
/**
   * Specifies the type of the action.
   * - `"button"`: A standard button action.
   * - `"toggle"`: An action that can switch between active and inactive states.
   * - `"expand-toggle"`: An action specifically for expanding or collapsing content.
   */
```

I prefer the simpler value of `type: "expand"` but would like your thoughts.

**Comment by geospatialem** [comment]
> I prefer the simpler value of `type: "expand"` but would like your thoughts.

Simple is always best 🏆 , also favor `type` of `"expand"`.

**Comment by driskull** [comment]
@geospatialem this one is installed, it just needs to be documented for 4.0.

**Comment by geospatialem** [comment]
Some verification considerations/helpers ✅ :
- [x] For functionality, refer to `action`'s `aria` property
- [x] For doc, refer to the internal doc `summary-notes-preview` branch/PR for:
    - [x] the `action` component page's `aria` property
    - [x] the `action` component page's accessibility section and
    - [x] summary notes

**Comment by isaacbraun** [comment]
Verified the presence and functionality of the `aria` property on `5.0.0-next.89` via https://codepen.io/isaacbraun/pen/MYeGBRp.

Additionally, verified that the `aria` property is publicly documented on the `next` site and the `summary-notes-preview` branch features the accessibility section and summary notes section 👍
